jekyll/_posts/2021-09-25-version-160-released.md

```diff
@@ -0,0 +1,891 @@
+  provide support for Unicode and do better job for avoiding margin overflows.
+- The RST parser now supports footnotes, citations, admonitions, short style references with symbols.
+- The RST parser now supports Markdown table syntax.
+  Known limitations:
+  - cell alignment is not supported, i.e. alignment annotations in a delimiter
+    row (`:---`, `:--:`, `---:`) are ignored,
+  - every table row must start with `|`, e.g. `| cell 1 | cell 2 |`.
+- Implemented `doc2tex` compiler command which converts documentation in
+  `.nim` files to Latex.
+- docgen now supports syntax highlighting for inline code.
+- docgen now supports same line doc comments:
+```nim
+func fn*(a: int): int = 42  ## Doc comment
+```
+- docgen now renders deprecated and other pragmas.
+- `runnableExamples` now works with templates and nested templates.
+- `runnableExamples: "-r:off"` now works for examples that should compile but not run. 
+- `runnableExamples` now renders code verbatim, and produces correct code in all cases.
+- docgen now shows correct, canonical import paths in docs.
+- docgen now shows all routines in sidebar, and the proc signature is now shown in sidebar.
+
+
+## Effects and checks
+- Significant improvement to error messages involving effect mismatches
+- There is a new `cast` section `{.cast(uncheckedAssign).}: body` that disables some
+  compiler checks regarding `case objects`. This allows serialization libraries
+  to avoid ugly, non-portable solutions. See RFC [#407](https://github.com/nim-lang/RFCs/issues/407).
+  for more details.
+
+
+## Tools
+- Major improvements to `nimgrep`, see PR [#15612
+](https://github.com/nim-lang/Nim/pull/15612).
+- `fusion` is now un-bundled from Nim, `./koch fusion` will
+  install it via Nimble at a fixed hash.
+- `testament`: added `nimoutFull: bool` spec to compare full output of compiler
+  instead of a subset; many bugfixes to testament.
+
+
+## Misc/cleanups
+- Deprecated `TaintedString` and `--taintmode`.
+- Deprecated `--nilseqs` which is now a noop.
+- Added `-d:nimStrictMode` in CI in several places to ensure code doesn't have certain hints/warnings.
+- Removed `.travis.yml`, `appveyor.yml.disabled`, `.github/workflows/ci.yml.disabled`.
+- `[skip ci]` now works in azure and CI pipelines, see detail in PR [#17561](https://github.com/nim-lang/Nim/pull/17561).
+
+
+# Footnotes
+Tested on a 2.3 GHz 8-Core Intel Core i9, 2019 macOS 11.5 with 64GB RAM.
+* [1] command used: `nim c -o:/tmp/main -d:danger --eval:'echo "hello world"'`
+* [2] commands used:
+  for nim: `nim c --forceBuild compiler/nim`
+  for rust: `./x.py build`, see also https://www.reddit.com/r/rust/comments/76jq7h/long_time_to_compile_rustc/
+  for gcc: see https://unix.stackexchange.com/questions/421822/how-long-does-it-take-to-compile-gcc-7-3-0
+  https://solarianprogrammer.com/2016/10/07/building-gcc-ubuntu-linux/
+  for clang: see https://quuxplusone.github.io/blog/2018/04/16/building-llvm-from-source/
+  for go: `./make.bash`
```

Comment:
```suggestion
  for Nim: `nim c --forceBuild compiler/nim`
  for Rust: `./x.py build`, see also https://www.reddit.com/r/rust/comments/76jq7h/long_time_to_compile_rustc/
  for gcc: see https://unix.stackexchange.com/questions/421822/how-long-does-it-take-to-compile-gcc-7-3-0
  https://solarianprogrammer.com/2016/10/07/building-gcc-ubuntu-linux/
  for clang: see https://quuxplusone.github.io/blog/2018/04/16/building-llvm-from-source/
  for Go: `./make.bash`
```

Reply:
Also `GCC` and `Clang`

Reply:
```suggestion
  for nim: `nim c --forceBuild compiler/nim`
  for rust: `./x.py build`, see also https://www.reddit.com/r/rust/comments/76jq7h/long_time_to_compile_rustc/
  for GCC: see https://unix.stackexchange.com/questions/421822/how-long-does-it-take-to-compile-gcc-7-3-0
  https://solarianprogrammer.com/2016/10/07/building-gcc-ubuntu-linux/
  for Clang: see https://quuxplusone.github.io/blog/2018/04/16/building-llvm-from-source/
  for go: `./make.bash`
```

in addition to the comment by konsumlamm